src/seedu/addressbook/Main.java

```diff
@@ -81,6 +81,12 @@ private void runCommandLoopUntilExitCommand() {
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
+            try {
+            	storage.checkFileExistOnDisk();
+            } catch (StorageFileDeletionException e) {
+            	ui.showToUser(e.getMessage());
+            	exit();
+            }
```

Comment:
This method may be too high-level to detect this problem. This method doesn't even know about a file. Perhaps you can detect it inside the `StorageFile.save()` instead?